tchannel/tornado/peer.py

```diff
@@ -656,6 +677,34 @@ def _update_heap(self, peer):
         peer.rank = rank
         self.peer_heap.update_peer(peer)
 
+    def get_for_request(self, hostport):
+        """Get a Peer for the given destination for a request.
+
+        A new Peer is added and returned if one does not already exist for the
+        given host-port. Otherwise, the existing Peer is returned.
+
+        **NOTE** new peers will not be added to the peer heap.
+        """
+        assert hostport, "hostport is required"
+        if hostport not in self._peers:
+            self.add_direct(hostport)
+
+        return self._peers[hostport]
+
+    def add_direct(self, hostport):
```

Comment:
Should all the new methods be underscore prefixed since I don't think they should be part of the public API surface